source/memory_test.go

```diff
@@ -0,0 +1,317 @@
+// Copyright 2021 The Witness Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package source
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/testifysec/go-witness/attestation"
+	"github.com/testifysec/go-witness/dsse"
+	intoto "github.com/testifysec/go-witness/intoto"
+)
+
+func TestLoadEnvelope(t *testing.T) {
+	// Marshal the attestation.Collection into a JSON byte array
+	predicate, err := json.Marshal(attestation.Collection{})
```

Comment:
I think that making this predicate a populated `attestation.Collection{}` might be nice here.

Reply:
Thinking that this should be more along the lines of the `validPredicate` var in `TestSearch`